### PR TITLE
Refactor VM node selector scenario

### PR DESCRIPTION
**Pull request by rhrazdil**
- Created model for Node Selector modal dialog
- updated node-selector.scenario.ts

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.node-selector.scenario.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 import { VirtualMachine } from './models/virtualMachine';
 import { getVMManifest } from './utils/mocks';
 import { getRandStr } from './utils/utils';
+import { MatchLabels } from '@console/internal/module/k8s';
 
 describe('KubeVirt VM detail - edit Node Selector', () => {
   const testVM = getVMManifest('Container', testName, `node-selector-vm-${getRandStr(5)}`);
-  const vm = new VirtualMachine(testVM.metadata);
+  const vm: VirtualMachine = new VirtualMachine(testVM.metadata);
+  const labels: MatchLabels = {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.node-selector.scenario.ts:14]: I would suggest to add at least two labels for this tests to make sure it works well. And then refactor `deleteLabel` to `deleteLabels`.

Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.node-selector.scenario.ts:14]: updated, adding and removing 2 key:value labels, added `deleteNodeSelectors` method.

Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.detail.node-selector.scenario.ts:14]: Looks good.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts [hunk 2]
```diff
@@ -39,4 +53,28 @@ export class VirtualMachine extends BaseVirtualMachine {
       timeout,
     );
   }
+
+  async addNodeSelectors(labels: MatchLabels) {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts:57]: Is it better to move `addNodeSelectors` to `kubevirtDetailView.ts`?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export class NodeSelectorDialog {
+  private extractID(id: string) {
+    return id.split('-')[1];
+  }
+
+  /**
+   * Adds new row if needed.
+   * Returns string with index of next empty row.
+   */
+  async addRow(): Promise<string> {
```
Comment (gouyang) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts:16]: Other place like `Tolerations` and `affinity` may use `addRow` as well, then we can move it to `utils` or `console-shared`.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts (hunk 2 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts:57]: No, scheduling is relevant for Virtual Machine only, it cannot be edited in VMI or Virtual Machine Templates

File: frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts (hunk 3 above)
Comment (rhrazdil) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts:16]: Better place would probably be some scheduling.dialog.view.ts file, utils and console-shared are for more general purpose utility methods

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts (hunk 2 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/models/virtualMachine.ts:57]: okay.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts (hunk 3 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/dialogs/nodeSelectorDialog.ts:16]: okay, if you decide not to do it in this one, then it looks good to me.

**Comment by rhrazdil** [comment]
@suomiy can you review please?

**Pull request merged**